### ISSUE TITLE
Are their instructions on how to get the setup working on a non virtualbox vm?

### ISSUE THREAD

**Issue by dylan-chong** [dictation-toolbox/aenea#169]
Virtual box on a 2018 MacBook Pro seems to have heaps of problems - it will crash many times a day, which is rather annoying as this can corrupt the Dragon profile and I have to restore from a snapshot.

Many months ago I attempted to use VM Ware, but could not get the host only adapter working. I gave up, and ended up deleting that virtual machine at some point in the past. 

I am willing to give it a go if there is someone who has tried to set things up on a non-Virtualbox virtual machine before, who I can ask for help.

**Comment by calmofthestorm** [comment]
The host only adapter is recommended for security, but not strictly required, I believe. It'd be hard to walk through how to set it up with a different networking setup, as I must confess I'm quite ignorant of computer networking:-)

I've also used Aenea with a VM running under qemu, for what it's worth. Not sure if that's on mac these days.

**Comment by dylan-chong** [comment]
> I'm quite ignorant of computer networking:-)

me too

> I've also used Aenea with a VM running under qemu, for what it's worth. Not sure if that's on mac these days.

qemu from memory is used by android studio for the android emulator on Mac OS, but was extremely slow!

**Comment by calmofthestorm** [comment]
On linux, Qemu can either use KVM for hardware acceleration, or do it's own thing not taking advantage of virtualization extensions. It'll be painfully slow (IMO unusable) without KVM, but works just fine with it. KVM is linux specific, but there may be some analog on other OS.

**Comment by dylan-chong** [comment]
I can't find anything particularly obvious on Google, so that may indicate qemu is not used very much on Mac OS, which probably won't be good for indicating good reliability...

**Comment by calmofthestorm** [comment]
At risk of exposing my ignorance of computer networking to the entire world, the only thing you strictly need to get Aenea working is a way for the client (in the VM) to connect to the server (on the host). If the guest has an IP address it can talk to the host on, it can connect directly. If it doesn't, but the host can talk to the guest, you can SSH into the guest from the host with a reverse SSH tunnel, and Aenea will run over that. I have done this with a Windows guest, but I forget the exact incantation.

If neither can address the other, you may have a hard time:/ Qemu will let you forward ports on the command line for cases like this, but I'm not sure what VMWare/VirtualBox offer for that. There must be something; I assume this is a common need.

No pressure, but when you do get it working it would be great to get that at least briefly documented, either here or we could commit it to the repo, for others to reference.

**Comment by dylan-chong** [comment]
> No pressure, but when you do get it working it would be great to get that at least briefly documented, either here or we could commit it to the repo, for others to reference.

I will get around to that sometime, no guarantee when, by will be sure to write some documentation. I've also been meaning to improve the setup documentation as well, as I remember it being very hard to get everything set up.

**Comment by calmofthestorm** [comment]
Yeah, the setup is nightmareishly complex. Improvements to the docs are always welcome. Ideally we could ship a VM image but unfortunately copyright. Another reason open source is the future.

**Comment by seananderson** [comment]
I've been using it on VMware Fusion for a couple years now. I can't remember exactly what I did, but this is just to note that it is definitely possible. I know when I set up a new laptop from scratch I had one heck of a time getting it to connect to the VM image I copied over, but I did eventually get it working again. It does freeze on restoring from sleep maybe once or twice a week, but it's pretty usable. I don't have access to outside networking from the VM. I didn't do anything fancy to connect the server and client, but I definitely had to fiddle a bunch, finally got it working, and made sure to never touch it again.

**Comment by Danesprite** [comment]
I use QEMU/KVM and virt-manager for my Aenea setup instead of VirtualBox. I am in the process of putting together some documentation for this and should have it up in a new pull request soon. Most of the steps are the same, but there are some differences with networking, USB redirection (for the microphone) and other optional things I find useful like folder sharing between the host and guest.

@dylan-chong Admittedly I am late to commenting on this issue, but I had the same experience as you did with VirtualBox on Mac OS. It was completely unusable for me. I ended up using the `_aenea.py` grammar's server commands and SSH forwarding to use Aenea on my macbook from my Linux box instead. This isn't very portable of course, but it works. Did you ever come up with a solution?
